go.mod

```diff
@@ -1,11 +1,16 @@
 module github.com/aws/aws-lambda-go
 
-go 1.12
+go 1.18
```

Comment:
could this break any consumers still using older versions? There's no code changes in this package that require a new minimum version but am thinking about if module aware `go get` on version 1.12 -> 1.17 is gonna complain about this.

Reply:
guess it'll be fine 🤷‍♂️

testing:

pushed this: https://github.com/bmoffatt/test/blob/4225d22556144454019816fa24ec561f3b77924e/go.mod#L3



then did this:
```
~/test2
$ go version
go version go1.17.8 darwin/amd64
~/test2
$ cat main.go
package main

import (
	"log"

	"github.com/bmoffatt/test"
)

func main() {
	log.Print(test.Test())
}
~/test2
$ go mod init test-test-test
go: creating new go.mod: module test-test-test
go: to add module requirements and sums:
	go mod tidy
~/test2
$ go mod tidy
go: finding module for package github.com/bmoffatt/test
go: found github.com/bmoffatt/test in github.com/bmoffatt/test v0.0.0-20220329001020-4225d2255614
~/test2
$ go run ./main.go
2022/03/28 17:14:22 test
```